core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java

```diff
@@ -400,8 +400,11 @@ public static Expression convert(Expression operand, Type fromType,
       if (operand instanceof UnaryExpression) {
         UnaryExpression una = (UnaryExpression) operand;
         if (una.nodeType == ExpressionType.Convert
-            || Primitive.of(una.getType()) == toBox) {
-          return Expressions.box(una.expression, toBox);
+            && Primitive.of(una.getType()) == toBox) {
+          Primitive origin = Primitive.of(una.expression.type);
```

Comment:
Why change `||` to `&&`.

Reply:
According to the comment, "Eliminate primitive casts like Long.valueOf((long) x), Generate Long.valueOf(x)", we should satiesfy all these conditions.

Reply:
Thanks for the explanation.